fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/LoanDelinquencyDomainServiceImpl.java

```diff
@@ -61,60 +59,90 @@ public CollectionData getOverdueCollectionData(final Loan loan) {
-                            boolean isLoanTransactionIsOnOrBeforeLastInstallmentDueDate = isLatestInstallment
-                                    && (loanTransaction.getTransactionDate().isEqual(installment.getDueDate())
-                                            || loanTransaction.getTransactionDate().isBefore(installment.getDueDate()));
-                            if (isLoanTransactionIsOnOrAfterInstallmentFromDate && (isLoanTransactionIsBeforeNotLastInstallmentDueDate
-                                    || isLoanTransactionIsOnOrBeforeLastInstallmentDueDate)) {
-                                amountAvailable = amountAvailable.subtract(loanTransaction.getAmount());
-                                if (amountAvailable.compareTo(BigDecimal.ZERO) < 0) {
-                                    overdueSinceDate = loanTransaction.getTransactionDate();
-                                    break;
-                                }
-                            }
-                        }
                     }
                 } else if (!firstNotYetDueInstallment) {
                     log.debug("Loan Id: {} with installment {} due date {}", loan.getId(), installment.getInstallmentNumber(),
                             installment.getDueDate());
                     firstNotYetDueInstallment = true;
-                    amountAvailable = installment.getTotalPaid(loanCurrency).getAmount();
-                    log.debug("Amount available {}", amountAvailable);
-                    for (LoanTransaction loanTransaction : chargebackTransactions) {
-                        boolean isLoanTransactionIsOnOrAfterInstallmentFromDate = loanTransaction.getTransactionDate().isEqual(
-                                installment.getFromDate()) || loanTransaction.getTransactionDate().isAfter(installment.getFromDate());
-                        boolean isLoanTransactionIsBeforeInstallmentDueDate = loanTransaction.getTransactionDate()
-                                .isBefore(installment.getDueDate());
-                        boolean isLoanTransactionIsBeforeBusinessDate = loanTransaction.getTransactionDate().isBefore(businessDate);
-                        if (isLoanTransactionIsOnOrAfterInstallmentFromDate && isLoanTransactionIsBeforeInstallmentDueDate
-                                && isLoanTransactionIsBeforeBusinessDate) {
-                            log.debug("Loan CB Transaction: {} {} {}", loanTransaction.getId(), loanTransaction.getTransactionDate(),
-                                    loanTransaction.getAmount());
-                            amountAvailable = amountAvailable.subtract(loanTransaction.getAmount());
-                            if (amountAvailable.compareTo(BigDecimal.ZERO) < 0 && !overdueSinceDateWasSet) {
-                                overdueSinceDate = loanTransaction.getTransactionDate();
-                                overdueSinceDateWasSet = true;
-                            }
-                        }
```

Comment:
It is hard to see what got changed. Can you please explain the changes in a couple sentences?